include/revng/Model/Filters.h

```diff
@@ -46,11 +50,39 @@ auto unwrap(const Type &Value, CallableType &&Callable) {
 
 } // namespace detail
 
+//
+// Generic filters
+//
+
+constexpr auto DereferencePointers = std::views::filter([](auto *Ptr) {
+                                       return Ptr != nullptr;
+                                     })
+                                     | std::views::transform([](auto *T) {
+                                         return *T;
+                                       });
```

Comment:
The name does not reflect well that you're ignoring `nullptr`s. At least add docs.

Reply:
Uhm, but this is not just a filter. Maybe `model::DereferenceNonNull`?

Reply:
Technically this is a factored-out piece of `model::filter::Struct` and `model::filter::Union`. I thought about keeping it in the `detail` namespace, but somebody else might find it useful - so I exposed the API.

That is to say, I'm fine with any name and any placement, as long as I don't need to spell these bulky lambdas in more than one place :)

Reply:
Ah OK!